templates/swift/Sources/Client.swift.twig

```diff
@@ -363,7 +374,77 @@ open class Client {
         }
     }
 
-    private func randomBoundary() -> String {
+    func chunkedUpload<T>(
+        path: String,
+        headers: inout [String: String],
+        params: inout [String: Any?],
+        paramName: String,
+        convert: (([String: Any]) -> T)? = nil,
+        onProgress: ((Double) -> Void)? = nil,
+        completion: ((Result<T, {{ spec.title | caseUcfirst }}Error>) -> Void)? = nil
+    ) {
+        let file = params[paramName] as! File
+        let size = file.buffer.readableBytes
+
+        if size < Client.chunkSize {
+            call(
+                method: "POST",
+                path: path,
+                headers: headers,
+                params: params,
+                convert: convert,
+                completion: completion
+            )
+            return
+        }
+
+        var input = file.buffer
+        var offset = 0
+        var result = [String:Any]()
+        let group = DispatchGroup()
+
+        while offset < size {
+            let slice = input.readSlice(length: Client.chunkSize)
+                ?? input.readSlice(length: size - offset)
+            
+            params[paramName] = File(
+                name: file.name,
+                buffer: slice!
+            )
+            
+            headers["content-range"] = "bytes \(offset)-\(min((offset + Client.chunkSize) - 1, size))/\(size)"
+
+            group.enter()
+
+            call(
+                method: "POST",
+                path: path,
+                headers: headers,
+                params: params,
+                convert: { return $0 }
+            ) { response in
+                switch response {
+                case let .success(map):
+                    result = map
+                    group.leave()
+                case let .failure(error):
+                    completion?(.failure(error))
+                    return
+                }
+            }
+            
+            group.wait()
+
+            offset += Client.chunkSize
+            headers["x-appwrite-id"] = result["$id"] as? String
```

Comment:
If you could replace this string with the value from the spec, it would be great 👍